app/src/main/java/org/bostwickenator/ftpuploader/FtpClient.java

```diff
@@ -9,43 +9,58 @@
 import java.io.FileInputStream;
 import java.io.IOException;
 
+import static java.lang.Integer.parseInt;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_PASSIVE;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_PASSWORD;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_PORT;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_SERVER;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_TARGET_DIR;
+import static org.bostwickenator.ftpuploader.SettingsActivity.SETTING_USERNAME;
+import static org.bostwickenator.ftpuploader.SettingsStore.settingsStore;
+
 class FtpClient {
 
     private final FTPClient ftpClient = new FTPClient();
     private final static String EMPTY_STRING = "";
-    SettingsStore settingsStore = SettingsStore.getSettingsStore();
+    private SettingsStore settingsStore = settingsStore();
 
     public void connect() {
 
         try {
-            ftpClient.connect(getStringWithEmptyAsDefault(SettingsActivity.SETTING_SERVER), Integer.parseInt(getStringWithEmptyAsDefault(SettingsActivity.SETTING_PORT)));
-            ftpClient.login(getStringWithEmptyAsDefault(SettingsActivity.SETTING_USERNAME), getStringWithEmptyAsDefault(SettingsActivity.SETTING_PASSWORD));
-            if (settingsStore.getBoolean(SettingsActivity.SETTING_PASSIVE, false)) {
-                ftpClient.enterLocalPassiveMode();
+            this.ftpClient.connect(getStringWithEmptyAsDefault(SETTING_SERVER), parseInt(this.settingsStore.getString(SETTING_PORT, "21")));
```

Comment:
Couldn't you just get this as a integer from the start or does the store not provide a .getInt() ?

Reply:
Didn't change that. Is from original code. Could surely be refactored also...